inbox/sqlalchemy_ext/util.py

```diff
@@ -289,6 +292,49 @@ def generate_public_id():
 
 # Other utilities
 
+RE_SURROGATE_CHARACTER = re.compile(r"[\ud800-\udfff]")
+RE_SURROGATE_PAIR = re.compile(r"[\ud800-\udbff][\udc00-\udfff]")
+
+
+def utf8_encode(text, errors="strict"):
+    # type: (str, str) -> Tuple[bytes, int]
+    return text.encode("utf-8", errors), len(text)
+
+
+def utf8_surrogate_fix_decode(memory):
```

Comment:
This may also take `errors`, right? Which we would then pass to `binary.decode` in the first attempt.

Reply:
Yes, although the database driver won't call it this way, but let's add it for completeness.